src_c/simd_blitters_sse2.c

```diff
@@ -424,10 +328,7 @@ alphablit_alpha_sse2_argb_no_surf_alpha(SDL_BlitInfo *info)
                     temp = _mm_and_si128(temp, rgb_mask_128);
                     /* add alpha to RGB */
                     temp = _mm_add_epi16(mm_dst_alpha, temp);
-                    *dstp64 =
-                        (((Uint64)_mm_cvtsi128_si32(_mm_srli_si128(temp, 4)))
-                             << 32 |
-                         (Uint32)_mm_cvtsi128_si32(temp));
+                    STORE_M128_INTO_64(&temp, dstp64);
```

Comment:
I'm not clear about this diff... I'm in no way experienced around this bits of code, but the code before applying the macro looks complex and this macro clearly does not expand to that

Reply:
In your opening comment you mentioned
> I also found that there were two other intrinsics that are 100% equivalent and available on 32 bit.

I assume that refers to this (please correct me if it's not the case)

Reply:
they actually are equivalent, it is just messy. Original is  64 bit SIMD -> to 32 bit non-SIMD via two 32 bit stores OR'd together, with one bitshifted 32 bits one way to fit into 32 bits on the way out of the SIMD registers, then shifted back the other way to fit into a 64 bit int. They were basically in 32 bit int format just for the store instruction. 

For whatever reason, while you can have 64 bit ints on a 32 bit system, but not all the 64 bit SIMD instructions work on a 32 bit architecture. Thus this awkward dance I wrote originally because I thought there were no 64 bit store operations on 32 bit systems. But apparently I missed something as there actually _are_ some 64 bit load and store operations that work on 32 bit (hooray) - they just are a bit slower than the ones I was using for the 64 bit path. 

Thus this new macro that lets us just use the 64 bit code for everything rather than a terrible hacked up version of it on 32 bit.

Reply:
Oh, this makes sense now, thanks for explaining!